types/Types.go

```diff
@@ -0,0 +1,32 @@
+package types

```

Comment:
generally not desired to have a package for `common`, `base`, or `utils` and this feels like it fits in that rules. If you have to use one of these packages then it might mean your packages are too thin and this is just avoiding a dependency cycle. If not then cool put them where they need to be, but it could be a good way to help organize things.

Try and think in terms of "containment" rather than "category" when packaging stuff